src/Workspaces/Core/Portable/Classification/Classifier.cs

```diff
@@ -24,7 +24,7 @@ public static class Classifier
             TextSpan textSpan,
             CancellationToken cancellationToken = default)
         {
-            var semanticModel = await document.GetSemanticModelForSpanAsync(textSpan, cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
so a core thing about this change is that:

1. it is to reduce latency in typing (which is latency sensitive).
2. it works by caching a speculative model associated with a *method body*.

the latter is key.  after typing we expect completion to activate, and ask a ton of questions aobut the current method body by a lot of providers.  We cache the current speculative model based on the current method body and can continue to reuse that.  However, if classification wakes up it may start asking about anything *in view*, which means it could start asking about *other* method bodies, causing the one that hte user is editing to be tossed.

So we no longer do that.  We only want to use this for features that are directly interacting with typing, esp. in affecting latency or causing commit-pauses.

Reply:
This does not seem very maintainable. How do we automate the detection of misuse?

Reply:
Also, semantic classification during typing often only reclassifies the line being edited, which would fall into the prime category for using the speculative semantic model.

Reply:
Classification classifies at least the full screen, and usually a couple hundred lines beyond that. The chances of hitting multiple method bodies is high.

Reply:
> which would fall into the prime category for using the speculative semantic model.

A separate system can be built for that. In that case we would get a potential latency benefit, but we don't have the strict requirement to have the same latency needs as completion.